addons/frag/functions/fnc_fired.sqf

```diff
@@ -19,40 +19,70 @@
 //IGNORE_PRIVATE_WARNING ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_vehicle", "_gunner", "_turret"];
 TRACE_10("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_vehicle,_gunner,_turret);
 
-private _shouldAdd = GVAR(cacheRoundsTypesToTrack) get _ammo;
-if (isNil "_shouldAdd") then {
-    TRACE_1("no cache for round",_ammo);
-
-    //Read configs and test if it would actually cause a frag, using same logic as FUNC(pfhRound)
-    private _skip = getNumber (configFile >> "CfgAmmo" >> _ammo >> QGVAR(skip));
-    private _explosive = getNumber (configFile >> "CfgAmmo" >> _ammo >> "explosive");
-    private _indirectRange = getNumber (configFile >> "CfgAmmo" >> _ammo >> "indirectHitRange");
-    private _force = getNumber (configFile >> "CfgAmmo" >> _ammo >> QGVAR(force));
-    private _fragPower = getNumber (configFile >> "CfgAmmo" >> _ammo >> "indirecthit") * (sqrt (getNumber (configFile >> "CfgAmmo" >> _ammo >> "indirectHitRange")));
+if (_ammo isEqualTo "" || {isNull _projectile} ||
+    !(if (isNil "_gunner") then {local _unit} else {local _gunner}) ||
+    {_projectile getVariable [QGVAR(blacklisted), false]}) exitWith {
+    TRACE_2("bad ammo or projectile, or blackList",_ammo,_projectile);
+};
 
-    _shouldAdd = (_skip == 0) && {(_force == 1) || {_explosive > 0.5 && {_indirectRange >= 4.5} && {_fragPower >= 35}}};
+if (GVAR(spallEnabled) && {_ammo call FUNC(shouldSpall)}) then {
+    private _hitPartEventHandler = _projectile addEventHandler [
+        "HitPart", {
+            params ["_projectile", "_hitObject", "", "_posASL", "_velocity"];
 
-    if (GVAR(spallEnabled) && {!_shouldAdd}) then {
-        private _caliber = getNumber (configFile >> "CfgAmmo" >> _ammo >> "caliber");
-        if !(_caliber >= 2.5 || {(_explosive > 0 && {_indirectRange >= 1})}) exitWith {}; // from check in doSpall: line 34
-        TRACE_1("Won't frag, but will spall",_caliber);
-        _shouldAdd = true;
-    };
+            // starting v2.18 it may be faster to use the instigator EH parameter, the same as the second entry shotParents, to recreate _shotParent
```

Comment:
```suggestion
            // as of v2.18 it may be faster to use the instigator EH parameter, the same as the second entry shotParents, to recreate _shotParent
```

Reply:
As of implies 2.18 is out (it isn't)

Reply:
Had no clue there's a diff.

Reply:
Alright, I'm awake now and actually reading it, "as of" would be ok, I was thinking of "as from." Either would work, so I don't think it's worth another commit over it. The next other but I fix I'll reword that comment.

Reply:
Fair enough.